server/scsynth/SC_Graph.cpp

```diff
@@ -431,15 +431,28 @@ void Graph_Ctor(World *inWorld, GraphDef *inGraphDef, Graph *graph, sc_msg_iter
 	inGraphDef->mRefCount ++ ;
 }
 
+// Calls the unit's Ctor func.
+inline void InitializeUnit(Unit* unit)
+{
+	try {
+		(*unit->mUnitDef->mUnitCtorFunc)(unit);
```

Comment:
throwing an c++ exception from a shared library via a plain c interface and catching from another shared library or application is technically undefined behaviour.

depending on compiler and c++ runtime, it may work or fail. i'd advise against it